lib/steps/setupRedirects.js

```diff
@@ -35,12 +36,17 @@ const setupRedirects = () => {
   // less-specific routes (e.g., catch-all)
   const sortedRoutes = getSortedRoutes(nextRedirects.map(({ route }) => route));
 
+  // Create variable for tracking if there is a catch-all route
+  let hasCatchAllRoute = false;
+
   // Assemble redirects for each route
   sortedRoutes.forEach((route) => {
     const nextRedirect = nextRedirects.find(
       (redirect) => redirect.route === route
     );
 
+    if (isCatchAllRoute(route)) hasCatchAllRoute = true;
```

Comment:
Use this outside of the sortedRoutes.forEach loop: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some ?

```js
const hasCatchAllRoute = sortedRoutes.some(route => isCatchAllRoute(route))
```

That way it could be closer to the line where `hasCatchAllRoute` is used. Ideally, add a comment to the line, too!

Reply:
Actually, it does not matter whether a catch-all route exists, right? What really matters is whether a catch-all route exists at root-level. Perhaps the better way to test for this would be to do something like this?

```js
hasRootCatchAll = redirects.some(redirect => redirect.startsWith("/*"))
if (hasRootCatchAll) {
  const rootCatchAllIndex = redirects.findIndex(redirect => redirect.startsWith("/*"))
  redirects.splice(rootCatchAllIndex, 0, "/_next/*  /_next/:splat  200")
}
```

Reply:
makes sense yeah much better